lte/gateway/c/core/oai/test/amf/BUILD.bazel

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
+
+package(default_visibility = ["//lte/gateway/c/core/test:__subpackages__"])
+
+cc_test(
+    name = "amf_algorithm_selection_test",
+    size = "small",
+    srcs = [
+        "test_amf_algorithm_selection.cpp",
+    ],
+    deps = [
+        "//lte/gateway/c/core",
+        "@com_google_googletest//:gtest_main",
+    ],
+)
+
+cc_test(
+    name = "amf_map_test",
+    size = "small",
+    srcs = [
+        "test_amf_map.cpp",
+    ],
+    deps = [
+        "//lte/gateway/c/core",
+        "@com_google_googletest//:gtest_main",
+    ],
+)
+
+cc_library(
+    name = "amf_app_test_util",
+    srcs = [
+        "amf_app_test_util.cpp",
+        "util_nas5g_auth_fail_pkt.cpp",
+        "util_nas5g_registration_pkt.cpp",
+        "util_nas5g_security_mode_reject_pkt.cpp",
+        "util_nas5g_service_request_pkt.cpp",
+        "util_nas5g_ul_nas_pdu_decode.cpp",
+        "util_s6a_update_location.cpp",
+    ],
+    hdrs = [
+        "amf_app_test_util.h",
```

Comment:
@electronjoe Running in bazel-base, I'm getting this
```bash
In file included from lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp:13:
./lte/gateway/c/core/oai/test/amf/amf_app_test_util.h:14:10: fatal error: gmock/gmock.h: No such file or directory
   14 | #include <gmock/gmock.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
INFO: Elapsed time: 280.477s, Critical Path: 17.42s
```

Reply:
Darn! Didn't notice this comment and merged.  I'll put up a PR now to address.

Reply:
Posted #12136